microceph/ceph/client_config.go

```diff
@@ -0,0 +1,61 @@
+package ceph
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microceph/microceph/common"
+	"github.com/canonical/microceph/microceph/database"
+)
+
+type ClientConfigT struct {
```

Comment:
ClientConfigT holds all the client configurations values that could be applicable for a host. This helps in keeping things neat and all the client config logic tied together. I believe using the native names would not save us complexity as:
1. As long as the field names are 'strings' reflection would be needed anyways, so using native names only saves us one function.
2. The same set is also used to provide the allowList (slice of config keys from the set) so the set is needed at multiple locations.
3. Since the key names follow snake case (as opposed to camel case) using them as  field names goes against the convention. (e.g. rbd_cache vs isCache)

Reply:
Thanks for the clarification. Would you please add a summary of the rationale (what the type is used for etc.) in the doc comment?

Reply:
Ack